### PR TITLE
Chore: provider file UTs

### PR DESCRIPTION
**Pull request by yaronya**
### Issue & Steps to Reproduce / Feature Request
Missing tests for the `env0/provider.go` file

### Solution
1. Fixed provider schema and set api key + api secret fields as `Required` and removed redundant assertion in `ConfigureFunc` method.
2. Used non-deprecated `ConfigureFuncContext` func instead of `ConfigureFunc`
3. Add proper tests
4. Regenerate docs for required api key + secret

### CODE REVIEW

File: env0/provider_test.go [hunk 1]
```diff
@@ -26,14 +32,83 @@ var testUnitProviders = map[string]func() (*schema.Provider, error){
...
 func runUnitTest(t *testing.T, testCase resource.TestCase, mockFunc func(mockFunc *client.MockApiClientInterface)) {
 	testReporter := utils.TestReporter{T: t}
-
 	ctrl = gomock.NewController(&testReporter)
+	defer ctrl.Finish()
+
+	os.Setenv("ENV0_API_KEY", "value")
+	os.Setenv("ENV0_API_SECRET", "value")
+	defer os.Setenv("ENV0_API_KEY", "")
+	defer os.Setenv("ENV0_API_SECRET", "")
```
Comment (razbensimon) [inline env0/provider_test.go:41]: nice one

File: env0/provider_test.go [hunk 2]
```diff
@@ -26,14 +32,83 @@ var testUnitProviders = map[string]func() (*schema.Provider, error){
...
 
 	apiClientMock = client.NewMockApiClientInterface(ctrl)
 	mockFunc(apiClientMock)
 
 	testCase.ProviderFactories = testUnitProviders
 	resource.UnitTest(&testReporter, testCase)
+}
+
+func TestProvider(t *testing.T) {
```
Comment (razbensimon) [inline env0/provider_test.go:50]: wondering the reason why this function is Public and other tests are Private methods.

File: env0/provider_test.go [hunk 3]
```diff
@@ -26,14 +32,83 @@ var testUnitProviders = map[string]func() (*schema.Provider, error){
...
+	diags := Provider().Validate(&terraform.ResourceConfig{})
+	testExpectedProviderError(t, diags, expectedKey)
+}
+
+func testMissingConfig(t *testing.T, config map[string]interface{}, expectedKey string) {
+	diags := Provider().Validate(terraform.NewResourceConfigRaw(config))
+	testExpectedProviderError(t, diags, expectedKey)
+}
+
+func TestMissingConfigurations(t *testing.T) {
```
Comment (razbensimon) [inline env0/provider_test.go:85]: here also

**Review by razbensimon — APPROVED** [review]
Amazing 💯

File: env0/provider_test.go (hunk 2 above)
Comment (yaronya) [reply env0/provider_test.go:50]: That's how `go` runs tests. 
When you do `go test`, go tests for every function that its name has a `Test` prefix and its signature is `t *testint.T`.

Not sure what's the private methods you refer to 🤔

File: env0/provider_test.go (hunk 3 above)
Comment (yaronya) [reply env0/provider_test.go:85]: See [answer](https://github.com/env0/terraform-provider-env0/pull/78#discussion_r643967434)

**Pull request merged**